src/org/omegat/util/gui/UIDesignManager.java

```diff
@@ -72,9 +75,34 @@
  */
 public final class UIDesignManager {
 
+    private static List<IThemeLoader> themeLoaders = new ArrayList<>();
+
     private UIDesignManager() {
     }
 
+    public static List<String> getThemes() {
+        List<String> themes = new ArrayList<>();
+        // Add default themes
+        themes.add("Default");
+        // Add from plugins
+        themes.addAll(themeLoaders.stream().map(t -> t.getName()).collect(Collectors.toList()));
+        return themes;
+    }
+
+    public static void registerThemeloader(IThemeLoader loader) {
+        themeLoaders.add(loader);
+    }
+
+    public static boolean launchThemeLoader(String name) {
+        Optional<IThemeLoader> loader = themeLoaders.stream().filter(t -> name.equals(t.getName())).findFirst();
+        if (loader.isPresent()) {
+            loader.get().onStartup();
```

Comment:
We need to add plugin's jar file to system class path where UIManager can see LaF library.

Loading external class to ancestor of class loader in dynamic manner is hack for Java8. After Java9, we need a custom class loader and specify it when launching JVM.